### ISSUE TITLE
Copy/Paste on Wayland (Sway)

### ISSUE THREAD

**Issue by concatime** [kovidgoyal/kitty#855]
Hi. It may seem a dumb question, but how do you copy/paste on Wayland?
I cant't right-click copy, nor Ctrl-Shit-c.
Do I need an external tool? From X11?
Thank you.

**Comment by concatime** [comment]
Yes, I was forcing Wayland. `glfw error 65544]: Wayland: Clipboard setting not implemented yet`. Do you know when it will be implemented?
Also, if I do not force wayland (default XWayland), I'm not able to type with my keyboard on kitty terminal. Is this a known bug?

**Comment by kovidgoyal** [comment]
See https://github.com/glfw/glfw/projects/2

The keyboard works fine under Xwayland (tested with weston)

**Comment by concatime** [comment]
(KITTY_ENABLE_WAYLAND=1)
Copying works fine, thank you.
But, copying from kitty, and pasting into itself does not work:
```
[248 09:31:15.922963] [glfw error 65544]: Wayland: Failed to read clipboard data from pipe (timed out)
[248 09:31:15.923122] [glfw error 65544]: Wayland: Could not copy writing to destination fd failed with error: Broken pipe
```
and playing with copy/paste crashes kitty:
```
listener function for opcode 2 of wl_data_source is NULL
fish: “kitty” terminated by signal SIGABRT (Abort)
```

**Comment by kovidgoyal** [comment]
The copy to self issue is easy to fix, but what does playing with copy/paste eman? Can you provide some set of steps to reproduce?

**Comment by concatime** [comment]
Try to copy from kitty to another app, and then from the other app, copy something back to kitty, one or multiple times.

**Comment by kovidgoyal** [comment]
What app are you using to cut from?

**Comment by concatime** [comment]
Chromium. Under Sway 1.0 Alpha.

**Comment by concatime** [comment]
Even under weston. Open weston terminal, type kitty and enter. Open another terminal. Copy from kitty to this new terminal, and then select something on this terminal and Ctrl-Shift-C will crash kitty.

**Comment by kovidgoyal** [comment]
Should be fine now.

**Comment by concatime** [comment]
It works fine now, thank you for your amazing job!

**Comment by Moelf** [comment]
I'm using Sway + Kitty + Skype (which is XWayland I guess), I have `KITTY_ENABLE_WAYLAND=1`, if I select or select `C-Shift-c` something in Kitty, I have trouble pasting it into Skype.

My current work around is open another Kitty, open Vim, paste into vim (through `C-Shift-v`), and use vim to copy to `"+y` which would access 'system clipboard'. Then I can paste into Skype

**Comment by kovidgoyal** [comment]
Does it work with other XWayland clients? Or is it only skype?

**Comment by kovidgoyal** [comment]
I cannot reproduce that after my fix.
